### PR TITLE
Use php5 query builder

### PR DESCRIPTION
**Pull request by micgro42**

### CODE REVIEW

File: action/entry.php [hunk 1]
```diff
@@ -217,17 +217,17 @@ protected function validate(AbstractBaseType $type, $label, $data) {
...
         }
 
+        if(!blank($data)) {
+            try {
+                $type->validate($data);
+            } catch(ValidationException $e) {
+                msg($prefix . $e->getMessage(), -1);
+                $ok = false;
+            }
+        }
```
Comment (splitbrain) [inline action/entry.php]: Why did you move this outside the else? Wouldn't multivalue types be validated twice now? A non-empty array is !blank.

Comment (micgro42) [inline action/entry.php]: No, the ``if(is_array($data))`` has an ``return $ok`` at its end.

Comment (splitbrain) [inline action/entry.php]: Ah! I missed that.
